litestar/contrib/sqlalchemy/dto.py

```diff
@@ -44,6 +45,19 @@
 class SQLAlchemyDTO(AbstractDTO[T], Generic[T]):
     """Support for domain modelling with SQLAlchemy."""
 
+    sqlalchemy_dto_config: SQLAlchemyDTOConfig
```

Comment:
That would. be a breaking change though. Would that be accepted for `v2.1`? 

Other than the breaking change, I think it makes sense since these configs can be considered as "advanced usage" that wouldn't need to be modified most of the time thus having a separate structure.

Reply:
Why would that be a breaking change?

Reply:
If someone is currently using the following:
```python
class MyModelDTO(SQLAlchemyDTO[MyModel]):
    config = DTOConfig()
```

And inside `SQLAlchemyDTO` we assume that the config is `SQLAlchemyDTOConfig`, that will raise an exception when accessing any attribute defined in `SQLAlchemyDTOConfig` hence everyone configuring `SQLAlchemyDTO` must migrate to `SQLAlchemyDTOConfig`

Reply:
> And inside `SQLAlchemyDTO` we assume that the config is `SQLAlchemyDTOConfig`

Or we could just not assume that :slightly_smiling_face: 

`SQLAlchemyDTO` should be able to support both `DTOConfig` and the specialised `SQLAlchemyDTOConfig`.

Reply:
```python
if isinstance(cls.config, DTOConfig):
    include_implicit_fields = True 
    # or SQLAlchemyDTOConfig().include_implicit_fields
    # or the config can be adapted to SQLAlchemyDTOConfig on init?
elif isintance(cls.config, SQLAlchemyDTOConfig):
    include_implicit_fields = cls.config.include_implicit_fields
```

Reply:
Adapting `DTOConfig` (if used) to `SQLAlchemyDTOConfig` on init looks like a good approach, shall I go with that and use a subclass?

Reply:
Looks good, yes!